cmd/print_test.go

```diff
@@ -58,3 +75,146 @@ func Test_getResourcePrinter(t *testing.T) {
 
 	}
 }
+
+func Test_getNamespaceFilter(t *testing.T) {
+	testCases := []struct {
+		name                      string
+		namespace                 string
+		allNamespaces             bool
+		expectedNamespaceFilter   string
+		expectingError            bool
+		expectingCurrentNamespace bool
+	}{
+		{
+			name:                      "Only namespace is specified",
+			namespace:                 "default",
+			allNamespaces:             false,
+			expectedNamespaceFilter:   "default",
+			expectingError:            false,
+			expectingCurrentNamespace: false,
+		},
+		{
+			name:                      "All namespaces overrides a specific namespace",
+			namespace:                 "default",
+			allNamespaces:             true,
+			expectedNamespaceFilter:   "",
+			expectingError:            false,
+			expectingCurrentNamespace: false,
+		},
+		{
+			name:                      "Current namespace used when nothing specified",
+			namespace:                 "",
+			allNamespaces:             false,
+			expectedNamespaceFilter:   "_",
+			expectingError:            false,
+			expectingCurrentNamespace: true,
+		},
+	}
+
+	destroy, err := setupKubeConfig()
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer destroy()
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			actualNamespaceFilter, err := getNamespaceFilter(tc.namespace, tc.allNamespaces)
+
+			if tc.expectingError && err == nil {
+				t.Errorf("Expected error but got none")
+			}
+			if !tc.expectingError && err != nil {
+				t.Errorf("Expected no error but got %v", err)
+			}
+
+			if tc.expectingCurrentNamespace {
+				tc.expectedNamespaceFilter, _ = getNamespaceFilter(tc.namespace, tc.allNamespaces)
+			}
+
+			if actualNamespaceFilter != tc.expectedNamespaceFilter {
+				t.Errorf(`getNamespaceFilter("%s", %v) = %v, expected %v`,
+					tc.namespace, tc.allNamespaces, actualNamespaceFilter, tc.expectedNamespaceFilter)
+			}
+		})
+
+	}
+}
+
+func setupKubeConfig() (func(), error) {
+	const kubeConfigPath = "/tmp/i2gw/.kube"
```

Comment:
prefer to use something like https://pkg.go.dev/os#MkdirTemp or https://pkg.go.dev/os#CreateTemp to create temp dirs/files for tests

Reply:
Good idea, thanks